vignettes/function-flow.Rmd

```diff
@@ -95,63 +95,202 @@ outputs.](images/function-flow-population.png)
+internal variables:
+
+-   LPR2-data:
+    -   `pnr`: identifier variable
+    -   `do_diagnosis`: include all diabetes diagnoses, registered as
+        primary (A) or secondary (B) diagnoses, regardless of type or
+        department: `c_diag` starts with "DE1[0-4]", "249", or "250" and
+        `c_diagtype` is either "A" or "B"
+    -   `is_primary`: Define whether the diagnosis was a primary
+        diagnosis (`c_diagtype` == "A")
+    -   `is_t1d`: Define whether the diagnosis was T1D-specific
+        (`c_diag` starts with "DE10" or "249")
+    -   `is_t2d`: Define whether the diagnosis was T2D-specific
+        (`c_diag` starts with "DE11" or "250")
+    -   `department`: Define whether the diagnosis was made made by an
+        endocrinological (`c_spec` == 8) or other medical department
+        (`c_spec` \< 8 or 9-30)
+-   LPR3:
+    -   `pnr`: identifier variable
+    -   `do_diagnosis`: include all diabetes diagnoses, registered as
+        primary (A) or secondary (B) diagnoses, regardless of type or
+        department: `diagnosekode` starts with "DE1[0-4]" and
+        `diagnosetype` is either "A" or "B", but exclude retracted
+        diagnoses (`senere_afkraeftet` == "Ja")
+    -   `is_primary`: Define whether the diagnosis was a primary
+        diagnosis (`diagnosetype` == "A")
+    -   `is_t1d`: Define whether the diagnosis was T1D-specific
+        (`diagnosekode` starts with "DE10")
+    -   `is_t2d`: Define whether the diagnosis was T2D-specific
+        (`diagnosekode` starts with "DE11")
+    -   `department`: Define whether the diagnosis was made made by an
+        endocrinological (`hovedspeciale_ans` == "medicinsk
+        endokrinologi") or other medical department (`hovedspeciale_ans`
+        either "Blandet medicin og kirurgi", "Intern medicin",
+        "Geriatri", "Hepatologi", "Hæmatologi", "Infektionsmedicin",
+        "Kardiologi", "Medicinsk allergologi", "Medicinsk
+        gastroenterologi", "Medicinsk lungesygdomme", "Nefrologi",
+        "Reumatologi", "Palliativ medicin", "Akut medicin",
+        "Dermato-venerologi", "Neurologi", "Onkologi", "Fysiurgi", or
+        "Tropemedicin")
+
+These intermediate results are combined for further processing, and
+`include_diabetes_diagnoses()` outputs a single `data.frame` with the
+following variables (one row for each individual):
+
+-   identifier variable (`pnr`)
+-   date of the first diabetes diagnosis (`do_diagnosis_1`)
+-   date of the second diabetes diagnosis (`do_diagnosis_2`)
+-   number of type 1 diabetes-specific primary diagnosis codes from
+    endocrinological departments (`n_t1d_endo`)
+-   number of type 2 diabetes-specific primary diagnosis codes from
+    endocrinological departments (`n_t2d_endo`)
+-   number of type 1 diabetes-specific primary diagnosis codes from
+    medical departments (`n_t1d_medical`)
+-   number of type 2 diabetes-specific primary diagnosis codes from
+    medical departments (`n_t2d_medical`)
```

Comment:
Are these all necessary at this point? Or could we have several columns called: `is_primary` (TRUE or FALSE), `diabetes_type` ("t1d" or "t2d"), `department` ("endocrinology" or "medical"), and `diagnosis_count` (number in each group)?

Reply:
For inclusion, only the dates of the first and second diagnosis is required, so all these are just for classification of diabetes type.
I think this is the minimal amount of variables that need to be passed further along for the type classification... the columns I wrote would be generated on a per-individual basis using the information found in the columns you suggested (or a by a logical statement to derive them directly without adding intermediate columns).